lib/conjur/api/patches/role.rb

```diff
@@ -0,0 +1,10 @@
+require 'conjur/api'
+
+class Conjur::Role
+  def can_admin_role? role
+    return true if self.roleid == role.roleid
+    
+    memberships = self.memberships.map(&:roleid)
+    role.members.find{|m| memberships.include?(m.member.roleid) && m.admin_option}
```

Comment:
This is confusing and deserves a comment.  Also, it might be more idiomatic to use `any?` to return a boolean value from a method ending in `?`.